src/stores/FavoritesController.ts

```diff
@@ -2,36 +2,43 @@ import { useAccountStore } from '#src/stores/AccountStore';
 import { getMediaItems, updatePersonalShelves } from '#src/stores/AccountController';
 import * as persist from '#src/utils/persist';
 import { useFavoritesStore } from '#src/stores/FavoritesStore';
-import type { Favorite } from '#types/favorite';
+import { useConfigStore } from '#src/stores/ConfigStore';
+import type { Favorite, SerializedFavorite } from '#types/favorite';
 import type { PlaylistItem } from '#types/playlist';
 
 const PERSIST_KEY_FAVORITES = `favorites${window.configId ? `-${window.configId}` : ''}`;
 
 export const restoreFavorites = async () => {
   const { user } = useAccountStore.getState();
+  const favorites_list = useConfigStore.getState().config.features?.favorites_list;
+
   const savedItems = user ? user.externalData?.favorites : persist.getItem<Favorite[]>(PERSIST_KEY_FAVORITES);
 
-  if (savedItems) {
-    const playlistItems = await getMediaItems(savedItems.map(({ mediaid }) => mediaid));
-    const favorites = playlistItems.map((item) => createFavorite(item));
+  if (savedItems?.length && favorites_list) {
+    const playlistItems = await getMediaItems(
+      favorites_list,
+      savedItems.map(({ mediaid }) => mediaid),
+    );
+
+    const favorites = (playlistItems || []).map((item) => createFavorite(item));
 
     useFavoritesStore.setState({ favorites });
   }
 };
 
-export const serializeFavorites = (favorites: Favorite[]) => {
-  return favorites.map(({ mediaid, title, tags, duration }) => ({ mediaid, title, tags, duration }));
+export const serializeFavorites = (favorites: Favorite[]): SerializedFavorite[] => {
+  return favorites.map(({ mediaid }) => ({ mediaid }));
 };
 
 export const persistFavorites = () => {
   const { favorites } = useFavoritesStore.getState();
   const { user } = useAccountStore.getState();
 
   if (user) {
-    return updatePersonalShelves();
```

Comment:
I think these are intentionally split. If your favorites save locally when you are logged in, it could leak personal data (i.e. if you favorite a bunch of embarrassingly bad movies, your partner would see them and make fun of you if they shared your browser.)

Reply:
Makes sense. I changed it back.